### PR TITLE
oh no it's a gun pr - adds the Scarborough Arms Enforcer-TEN, a 10mm handgun

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request

Adds the Scarborough Arms `Enforcer-TEN` handgun to... uh, nowhere, at the time of writing. I'll probably put it in the OPFOR menu/uplink eventually. It has a cool animated chamber indicator (leftmost light) *and* a visibly changing safety indicator (rightmost light)!

Credits to Baystation for their magnum pistol sprites (which I have modified slightly), heavy pistol shot sound, and the heavy pistol magazine insert/removal sounds.

As It Stands:
- 40 damage per shot (c10mm base damage) (4 shots to down unarmored)
- 10 (+1) capacity (uses its own magazines)
- 17 force 20 throwforce
- normal size
- it's renameable with a pen

Uplink Stats Subject To Change:
- 10 TC for the gun itself
- 1 TC for "basic" magazines (rubber/lethal)
- 2 TC for "advanced" magazines (HP/AP/incendiary)

## How This Contributes To The Skyrat Roleplay Experience
probably doesnt b/c shooting people isnt roleplay

## Why It's Good For The Game
adds a layer of force between the makarov (small 1911, 30 base damage) and the revolver (the designated "i am no longer role playing" handcannon, 60 damage)

makarov (30 damage) (5 shots to down raw)
This Gun (40 damage) (4 shots to down raw)
revolver (60 damage) (3 shots to down raw)

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

inhands + item
dreamseeker_2023-08-13_09-07-50-AM-Sunday
updated magazine bands
image
image
image

</details>

## Changelog

:cl:
add: Scarborough Arms is proud to announce that they're finally re-exporting the Enforcer-TEN 10mm handgun to their customers in the frontier; which is to say, not Nanotrasen. In the event that you see it, you should probably reconsider being in its general vicinity.
add: Reports from Syndicate double-agents indicate that the Enforcer is available for 10 telecrystals, and its ammo remains at a similar price point to its more concealable counterpart.
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
No GBP? Maintainers close this pr this instant.

File: modular_skyrat/modules/modular_weapons/code/pistol.dm [hunk 1]
```diff
@@ -40,6 +40,72 @@
...
+	if(chamber_indicator_overlay)
+		if(chambered)
+			. += "[icon_state]_chambered"
+	// safety indicator
+	if(safety_indicator_overlay)
+		var/datum/component/gun_safety/our_safety = GetComponent(/datum/component/gun_safety)
+		if(our_safety.safety_currently_on)
+			. += "[icon_state]_safety-on"
+		else
+			. += "[icon_state]_safety-off"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/pistol.dm]:
```suggestion

	// safety indicator
	if(safety_indicator_overlay)
		var/datum/component/gun_safety/our_safety = GetComponent(/datum/component/gun_safety)
		. += "[icon_state]_safety-[our_safety.safety_currently_on ? "on" : "off"]"
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
The color palette and the shading of the gun strike me as a little odd compared to what we already have in the game, no?

**Comment by Hatterhat** [comment]
> The color palette and the shading of the gun strike me as a little odd compared to what we already have in the game, no?

yeah that's fair i just couldn't think of anywhere else to steal a Big Chunky Handgun sprite from that... wasn't eris

**Pull request closed without merging**